src/gui/csvImport/CsvParserModel.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+ *  Copyright (C) 2016 Enrico Mariotti <enricomariotti@yahoo.it>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "CsvParserModel.h"
+
+CsvParserModel::CsvParserModel(QObject *parent)
+    : QAbstractTableModel(parent)
+    , m_skipped(0)
+{}
+
+CsvParserModel::~CsvParserModel()
+{}
+
+void CsvParserModel::setFilename(const QString& filename) {
+    m_filename = filename;
+}
+
+QString CsvParserModel::getFileInfo(){
+    QString a(QString::number(getFileSize()).append(tr(" byte, ")));
+    a.append(QString::number(getCsvRows())).append(tr(" rows, "));
+    a.append(QString::number((getCsvCols()-1))).append(tr(" columns"));
+    return a;
+}
+
+bool CsvParserModel::parse() {
+    bool r;
+    beginResetModel();
+    m_columnMap.clear();
+    if (CsvParser::isFileLoaded()) {
+        r = CsvParser::reparse();
+    }
+    else {
+        QFile csv(m_filename);
+        r = CsvParser::parse(&csv);
+    }
+    for (int i=0; i<columnCount(); i++) {
+        m_columnMap.insert(i,0);
+    }
+    addEmptyColumn();
+    endResetModel();
+    return r;
+}
+
+void CsvParserModel::addEmptyColumn() {
+    for (int i=0; i<m_table.size(); ++i) {
+        csvrow r = m_table.at(i);
+        r.prepend(QString(""));
+        m_table.replace(i, r);
+    }
+}
+
+void CsvParserModel::mapColumns(int csvColumn, int dbColumn) {
+    if ((csvColumn < 0) || (dbColumn < 0) ) {
+        return;
+    }
+
+    beginResetModel();
+    if (csvColumn >= getCsvCols()) {
+        m_columnMap[dbColumn] = 0; //map to the empty column
+    }
+    else {
+        m_columnMap[dbColumn] = csvColumn;
+    }
+    endResetModel();
+}
+
+void CsvParserModel::setSkippedRows(int skipped) {
+    m_skipped = skipped;
+    QModelIndex topLeft = createIndex(skipped,0);
+    QModelIndex bottomRight = createIndex(m_skipped+rowCount(), columnCount());
+    Q_EMIT dataChanged(topLeft, bottomRight);
```

Comment:
We decided on using MOC keywords now instead of Q_* macros, so use emit instead of Q_EMIT